src/Calculator.ts

```diff
@@ -116,20 +118,109 @@ export function calculate(
  * @param measureBundle Bundle with a MeasureResource and all necessary data for execution.
  * @param patientBundles List of bundles of patients to be executed.
  * @param options Options for calculation.
- * @returns MeasureReport resource for each patient.
+ * @returns MeasureReport resource for each patient according to standard https://www.hl7.org/fhir/measurereport.html
  */
 export function calculateMeasureReports(
   measureBundle: R4.IBundle,
   patientBundles: R4.IBundle[],
   options: CalculationOptions
 ): R4.IMeasureReport[] {
-  return [
-    {
-      resourceType: 'MeasureReport',
-      measure: 'Measure/ImplementMe',
-      period: {}
-    }
-  ];
+  // options should be updated by this call if measurementPeriod wasn't initially passed in
+  const results: ExecutionResult[] = calculate(measureBundle, patientBundles, options);
+  const reports: R4.IMeasureReport[] = [];
+  results.forEach(function (result) {
+    const report: R4.IMeasureReport = <any>{};
+
+    // simple fields
+    report.resourceType = 'MeasureReport';
+    report.period = {
+      start: options.measurementPeriodStart, // double check format of start and end that we're passing in https://www.hl7.org/fhir/datatypes.html#dateTime... we don't seem to be passing anything in from CLI
+      end: options.measurementPeriodEnd
+    };
+    report.status = R4.MeasureReportStatusKind._complete; // are there cases where this should be _pending or _error?
+    report.type = R4.MeasureReportTypeKind._individual; // are there cases where this should be _subjectList or _summary or _dataCollection
+
+    // measure url from measure bundle
+    const measureEntry = measureBundle.entry?.find(e => e.resource?.resourceType === 'Measure') as R4.IBundle_Entry;
+    const measure = measureEntry.resource as R4.IMeasure;
```

Comment:
```suggestion
    const measureEntry = measureBundle.entry?.find(e => e.resource?.resourceType === 'Measure');
    const measure = measureEntry?.resource as R4.IMeasure;
```

The type of the `find` will be implied as a bundle entry. It also might be undefined, so the "proper" way around this is to add the questionMark on the subsequent line for the measureEntry